dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -355,6 +369,29 @@ private void provisionInputFile(String parameterFilePath, String targetPath, Pat
         }
     }
 
+    /**
+     * Returns the first target path and its scheme in <code>targetPaths</code>
+     * whose scheme is handled by a plugin in <code>pluginList</code>.
+     *
+     * @param pluginList     a list of plugins
+     * @param targetPaths    a list of targetPaths
+     * @return the first targetPath and scheme with a supported scheme, or <code>Optional.empty()</code>
+     */
+    static Optional<ImmutablePair<String, String>> findSupportedTargetPath(List<ProvisionInterface> pluginList, List<String> targetPaths) {
+        for (String target : targetPaths) {
+            URI objectIdentifier = URI.create(target);    // throws IllegalArgumentException if it isn't a valid URI
+            String scheme = objectIdentifier.getScheme().toLowerCase();
+            if (scheme != null) {
```

Comment:
The variable scheme will never be null. If `objectIdentifier.getScheme()` returns null, if that's possible, then you'll get a null pointer exception on `toLowerCase()` and this code will never execute. If it doesn't return null, then `toLowerCase()` should never return null.

You should check to see if objectIdentifier.getScheme returns null.